src/webgpu/api/operation/vertex_state/index_format.spec.ts

```diff
@@ -0,0 +1,369 @@
+export const description = `Test indexing, index format and primitive restart.
+
+TODO(hao.x.li@intel.com): Test that use the primitive restart values as real indices with
+non-strip topologies, to make sure those behave properly (and primitive restart is appropriately
+enabled/disabled) across backends. These tests will be important for gpuweb/gpuweb#1220.
+`;
+
+import { params, poptions } from '../../../../common/framework/params_builder.js';
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { GPUTest } from '../../../gpu_test.js';
+import { getTextureCopyLayout } from '../../../util/texture/layout.js';
+
+const kHeight = 4;
+const kWidth = 4;
+const kTextureFormat = 'bgra8unorm' as const;
+const kValidPixelColor = new Uint8Array([0x00, 0xff, 0x00, 0xff]); // green
+const kInvalidPixelColor = new Uint8Array([0x00, 0x00, 0x00, 0xff]); // black
+
+/**
+ * The expected rendering in raster grids of 4x4 pixels:
+ * Points:
+ *  ------------
+ *  |#        #|
+ *  |          |
+ *  |          |
+ *  |#        #|
+ *  ------------
```

Comment:
I removed these comments because now that we have the literals, they're redundant. (They don't appear in the doc comment, unfortunately, but better to get rid of redundancy.)

Reply:
Ack.